reference/api/keys.md

```diff
@@ -1,30 +1,264 @@
+
+## Get all keys
 
 <RouteHighlighter method="GET" route="/keys"/>
 
-Get the **private** and **public** [key](/reference/features/authentication.md).
+Get information on all API keys.
+
+### Example
+
+<CodeSamples id="get_all_keys_1" />
+
+#### Response: `200 Ok`
+
+```json
+{
+  "results": [
+    {
+      "description": "Manage documents: Products/Reviews API key",
+      "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+      "actions": [
+        "documents.add",
+        "documents.delete"
+      ],
+      "indexes": [
+        "products",
+        "reviews"
+      ],
+      "expiresAt": "2021-12-31T23:59:59Z",
+      "createdAt": "2021-10-12T00:00:00Z",
+      "updatedAt": "2021-10-13T15:00:00Z"
+    },
+    {
+      "description": "Default Search API Key (Use it to search from the frontend code)",
+      "key": "0a6e572506c52ab0bd6195921575d23092b7f0c284ab4ac86d12346c33057f99",
+      "actions": [
+        "search"
+      ],
+      "indexes": [
+        "*"
+      ],
+      "expiresAt": null,
+      "createdAt": "2021-08-11T10:00:00Z",
+      "updatedAt": "2021-08-11T10:00:00Z"
+    },
+    {
+      "description": "Default Admin API Key (Use it for all other operations. Caution! Do not share it on the client side)",
+      "key": "380689dd379232519a54d15935750cc7625620a2ea2fc06907cb40ba5b421b6f",
+      "actions": [
+        "*"
+      ],
+      "indexes": [
+        "*"
+      ],
+      "expiresAt": null,
+      "createdAt": "2021-08-11T10:00:00Z",
+      "updatedAt": "2021-08-11T10:00:00Z"
+    }
+  ]
+}
+```
+
+To learn more about the fields displayed here, take a look at the [API key object definition](#api-key-object-definition).
+
+## Create a key
+
+<RouteHighlighter method="POST" route="/keys/:key"/>
+
+Create an API key with the provided description, permissions, and expiration date.
+
+Only the `indexes`, `actions`, and `expiresAt` fields are mandatory. For more information, take a look at the [API key object definition](#api-key-object-definition).
+
+### Example
+
+<CodeSamples id="create_a_key_1" />
+
+#### Response: `201 Created`
+
+```json
+{
+    "description": "Add documents: Products API key",
+    "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+    "actions": [
+        "documents.add"
+    ],
+    "indexes": ["products"],
+    "expiresAt": "2021-11-13T00:00:00Z",
+    "createdAt": "2021-11-12T10:00:00Z",
+    "updatedAt": "2021-11-12T10:00:00Z"
+}
+```
+
+To learn more about the fields displayed here, take a look at the [API key object definition](#api-key-object-definition).
+
+## Update a key
+
+<RouteHighlighter method="PATCH" route="/keys/:key"/>
 
-::: warning
-You must have the master key to access this route.
-:::
+Update the description, permissions, and/or expiration date of an API key.
+
+You do not have to reproduce the API key object in the payload. **Provide only the fields you intend to update.**
```

Comment:
I find the first sentence a bit confusing. Perhaps we can drop it?

Not sure this is any better:

```suggestion
This is a `PATCH` route, which means you can provide only the fields you intend to update.
```